foundation/plugins/org.jboss.tools.foundation.ui/src/org/jboss/tools/foundation/ui/util/BrowserUtility.java

```diff
@@ -66,4 +104,280 @@ public void openUrl(String url, IWebBrowser browser, String pluginId, ILog log)
+					String noBrowserText) {
+		return createBrowserOrLink(style, parent, preferredBrowser, new IURLProvider() {
+			public String getUrl() {
+				return url;
+			}
+		}, noBrowserText);
+	}
+	
+	public static Control createBrowserOrLink(int style, Composite parent, int preferredBrowser, IURLProvider url,
+			String noBrowserText) {
+
+		Control result = null;
+		Browser browser = createBrowser(style, parent, preferredBrowser);
+		if (browser == null) {
+			//TODO: Implement loading for .txt files and showing them in 
+			// link as a text with visible hyperlink to open original address
+			final Link link = new Link(parent, SWT.NONE);
+			link.setText(noBrowserText);
+			SelectionListener openExBrowser = new OpenExBrowserListener(url);
+			link.addSelectionListener(openExBrowser);
+			result = link;
+		} else {
+			result = browser;
+		}
+		return result;
+	}
+
+	private static void showProgressAndDownloadText(final Link link, final String URL) {
+		final String msg = MessageFormat.format("Loading text file from {0}", URL);
+		Job licenseDownloadJob = new Job("msg") {
+			@Override
+			protected IStatus run(IProgressMonitor monitor) {
+				URLTransportUtility transport = new URLTransportUtility();
+				try {
+					File propFile = transport.getCachedFileForURL(URL.toString(), msg,
+									URLTransportUtility.CACHE_FOREVER, (int) 30000, 30000,
+									new IProgressMonitor() {
+
+						@Override
+						public void worked(final int work) {
+							link.getDisplay().asyncExec(new Runnable() {
+
+								@Override
+								public void run() {
+									link.setText("Progress " + work);
+								}
+							});
+						}
+
+						@Override
+						public void subTask(String name) {
+							// TODO Auto-generated method stub
+
+						}
+
+						@Override
+						public void setTaskName(String name) {
+							// TODO Auto-generated method stub
+						}
+
+						@Override
+						public void setCanceled(boolean value) {
+							// TODO Auto-generated method stub
+						}
+
+						@Override
+						public boolean isCanceled() {
+							// TODO Auto-generated method stub
+							return false;
+						}
+
+						@Override
+						public void internalWorked(double work) {
+							// TODO Auto-generated method stub
+						}
+
+						@Override
+						public void done() {
+							link.getDisplay().asyncExec(new Runnable() {
+								@Override
+								public void run() {
+									// link
+								}
+							});
+						}
+
+						@Override
+						public void beginTask(String name, int totalWork) {
+							// TODO Auto-generated method stub
+						}
+					});
+				} catch (CoreException e) {
+
+				}
+
+				return null;
+			}
+		};
+
+	}
+
+	private static Menu createLinkDefaultContextMenu (Control link, String url) {
+		Menu popupMenu = new Menu(link);
+	    MenuItem refreshItem = new MenuItem(popupMenu, SWT.NONE);
+	    refreshItem.setText("Messages.DownloadRuntimeLicenseFragment_Open_in_external_browser");
```

Comment:
why is there anythign about runtime license fragment in the generic browser utility ?